contracts/contracts/coordination/subscription/BqETHSubscription.sol

```diff
@@ -145,7 +153,15 @@ contract BqETHSubscription is EncryptorSlotsSubscription, Initializable, Ownable
     }
 
     function baseFees() public view returns (uint256) {
-        return baseFeeRate * subscriptionPeriodDuration * maxNodes;
+        uint256 currentPeriodNumber = getCurrentPeriodNumber();
+        return baseFees(currentPeriodNumber);
+    }
+
+    function baseFees(uint256 periodNumber) public view returns (uint256) {
+        uint256 baseFeeRate = initialBaseFeeRate *
+            (INCREASE_BASE + baseFeeRateIncrease) ** periodNumber;
```

Comment:
Maybe we should add a note here, but this can overflow after 15 or 16 periods. I don't think it's a major concern, but at least a dev note would be OK. There's also iterative implementations that would remove the possibility of overflow, but they will increase code complexity, and it's probably not worth it.

Reply:
✔️